DMCompiler/DM/Visitors/DMProcBuilder.cs

```diff
@@ -668,12 +668,37 @@ public void ProcessStatementSet(DMASTProcStatementSet statementSet)
                     string caseLabel = _proc.NewLabelName();
 
                     foreach (DMASTExpression value in switchCaseValues.Values) {
-                        if (value is DMASTSwitchCaseRange range) {
+                        if (value is DMASTSwitchCaseRange range) { // if(1 to 5) or something
                             if (!DMExpression.TryConstant(_dmObject, _proc, range.RangeStart, out var lower))
                                 throw new CompileErrorException(new CompilerError(range.RangeStart.Location, "Expected a constant"));
                             if (!DMExpression.TryConstant(_dmObject, _proc, range.RangeEnd, out var upper))
                                 throw new CompileErrorException(new CompilerError(range.RangeEnd.Location, "Expected a constant"));
-
+                            //DM 514.1580 does NOT care if the constants within a range are strings, and does a strange conversion to 0 or something, without warning or notice.
+                            //We are deviating from parity here and just calling that a CompilerError.
+                            if(lower is not Number)
+                            {
+                                if (lower is Null) // We do a little null coercion, as a treat
+                                    {
+                                    lower = new Number(lower.Location, 0.0f);
+                                    DMCompiler.Warning(new CompilerWarning(range.RangeStart.Location, "Malformed range, lower bound is coerced from null to 0"));
+                                }
+                                else
+                                {
+                                    throw new CompileErrorException(new CompilerError(range.RangeStart.Location, "Invalid range, lower bound is not a number"));
```

Comment:
It would be better to use `DMCompiler.Error()` here and continue generating the bytecode for the switch case. Doing it this way leads to other possible errors not being found as well as invalid uses of labels (creating errors like `Invalid label "label1"`).